### PR TITLE
Change TreeLSTM to CNN

### PR DESCRIPTION
**Pull request by dliu-gallup**
Only change TreeLSTM to CNN. Haven't updated the parsing tree in 'Translating trees to discriminator' process.

### CODE REVIEW

File: src/gan/discriminator/model.py [hunk 1]
```diff
@@ -60,13 +63,19 @@ def __init__(self, args, vocab): #vocab_size, in_dim, mem_dim, sparsity, freeze)
...
-        self.childsumtreelstm = ChildSumTreeLSTM(self.in_dim, self.mem_dim)
-        self.confidence = nn.Linear(self.mem_dim, 1)
+        #self.childsumtreelstm = ChildSumTreeLSTM(self.in_dim, self.mem_dim)
+        
+        self.discriminator_cnn = nn.Sequential(
+                    nn.Conv1d(in_channels = self.in_dim, out_channels = 1,  
+                        kernel_size = 1, stride = 1, padding = 1),
+                    nn.ReLU(inplace = True),
+                    nn.BatchNorm1d(1)) 
+
```
Comment (gallupBenRyan) [inline src/gan/discriminator/model.py]: Diana,

This is a good start, but I think we will want to make the CNN implementation somewhat more complex to capture more about the longer range dependencies in the inputs. Checkout the `cnn_classifier()` class in our verbatim coding repo - it's a pytorch implementation of a classic CNN text classifier system that should work quite well (https://arxiv.org/abs/1408.5882 for the source paper).

Comment (dliu-gallup) [reply src/gan/discriminator/model.py]: Updated files on CNN_Discriminator branch.

**Comment by gallupBenRyan** [comment]
Okay, after much further merging of different streams of work, this is ready for a re-review. Tagging Matt and Jeremy for outsider eyes. This combines the upgrades/fixes to the policy gradient algorithm, the conversion of the Discriminator system to a CNN architecture, associated changes to datasets/inputs for that Discriminator, and some necessary updates and fixes identified in expanding the training dataset with 10x more new cleaned queries.

File: src/gan/discriminator/model.py [hunk 2]
```diff
@@ -1,72 +1,213 @@
...
+            loss = criterion(outputs, labels.argmax(1))
+            
+            # Backward and optimize
+            self.optimizer.zero_grad()
+            loss.backward()
+            self.optimizer.step()
+
+        return loss.item()
+
+            # self.eval()
```
Comment (jbiggsgallup) [inline src/gan/discriminator/model.py]: This is extremely minor, but if this whole block is being commented out, can it be removed? (I'm generally not a fan of including commented-out code. It's a little confusing.)

File: src/gan/discriminator/model.py [hunk 3]
```diff
@@ -1,72 +1,213 @@
...
+    def __init__(self, args, vocab, vectors, output_shape):
+        super(QueryGAN_Discriminator_CNN, self).__init__()
 
-        if tree.num_children == 0:
-            child_c = inputs[0].detach().new(1, self.mem_dim).fill_(0.).requires_grad_()
-            child_h = inputs[0].detach().new(1, self.mem_dim).fill_(0.).requires_grad_()
-        else:
-            child_c, child_h = zip(* map(lambda x: x.state, tree.children))
-            child_c, child_h = torch.cat(child_c, dim=0), torch.cat(child_h, dim=0)
+        self.args = args
```
Comment (jbiggsgallup) [inline src/gan/discriminator/model.py]: If the `args` dictionary contains required arguments (like the `filter_sizes`), maybe pass these separately to the constructor?
